app/src/main/java/io/homeassistant/companion/android/settings/vehicle/views/AndroidAutoFavoritesView.kt

```diff
@@ -0,0 +1,132 @@
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun AndroidAutoFavoritesSettings(
+    androidAutoViewModel: ManageAndroidAutoViewModel,
+    serversList: List<Server>,
+    defaultServer: Int
+) {
+    val reorderState = rememberReorderableLazyListState(
+        onMove = { from, to -> androidAutoViewModel.onMove(from, to) },
+        canDragOver = { draggedOver, _ -> androidAutoViewModel.canDragOver(draggedOver) },
+        onDragEnd = { _, _ -> androidAutoViewModel.saveFavorites() }
+    )
+
+    var selectedServer by remember { mutableStateOf(defaultServer) }
+
+    val favoriteEntities = androidAutoViewModel.favoritesList
+    var validEntities by remember { mutableStateOf<List<Entity<*>>>(emptyList()) }
+    LaunchedEffect(favoriteEntities.size, androidAutoViewModel.sortedEntities.size) {
+        validEntities = withContext(Dispatchers.IO) {
+            androidAutoViewModel.sortedEntities
+                .filter {
+                    !favoriteEntities.contains("$selectedServer-${it.entityId}") &&
+                        (it.entityId.split(".")[0] in MainVehicleScreen.SUPPORTED_DOMAINS)
+                }
+                .toList()
+        }
+    }
+
+    LazyColumn(
+        state = reorderState.listState,
+        contentPadding = PaddingValues(vertical = 16.dp),
+        modifier = Modifier
+            .reorderable(reorderState)
+    ) {
+        item {
+            Text(
+                text = stringResource(commonR.string.aa_set_favorites),
+                fontWeight = FontWeight.Bold,
+                modifier = Modifier.padding(horizontal = 16.dp)
+            )
+        }
+
+        if (serversList.size > 1) {
+            item {
+                ServerExposedDropdownMenu(
+                    servers = serversList,
+                    current = selectedServer,
+                    onSelected = {
+                        androidAutoViewModel.loadEntities(it)
+                        selectedServer = it
+                    },
+                    modifier = Modifier.fillMaxWidth().padding(all = 16.dp)
+                )
+            }
+        }
+        item {
+            SingleEntityPicker(
+                entities = validEntities,
+                currentEntity = null,
+                onEntityCleared = { /* Nothing */ },
+                onEntitySelected = {
+                    androidAutoViewModel.onEntitySelected(true, it, selectedServer)
+                    return@SingleEntityPicker false // Clear input
+                },
+                modifier = Modifier.padding(all = 16.dp),
+                label = { Text(stringResource(commonR.string.add_favorite)) }
+            )
+        }
+        if (favoriteEntities.isNotEmpty() && androidAutoViewModel.sortedEntities.isNotEmpty()) {
+            items(favoriteEntities.size, { favoriteEntities[it] }) { index ->
+                val favoriteEntity =
+                    favoriteEntities[index].replace("[", "").replace("]", "").split("-")
```

Comment:
Unless I'm missing something favorites never contain `[]`? Unless that is the fix for ordering favorites, just changing the string set into a string that holds the array like we do in the database (should actually work, entity IDs never contain `[],`).

Reply:
Its converted to a string so if its the first or last value, or only value it has the potential to be wrapped in `[]`

Reply:
Now that it is converting a string to a list in the prefs repository, which removes all `[],`, the replacing isn't needed.